pkg/service/service.go

```diff
@@ -356,51 +334,30 @@ func ListOperatorServices(client *kclient.Client) ([]unstructured.Unstructured,
 	return allCRInstances, failedListingCR, nil
 }
 
-// GetGVKRFromCR returns values for group, version, kind and resource for a
-// given Custom Resource (CR)
-func GetGVKRFromCR(cr olm.CRDDescription) (group, version, kind, resource string, err error) {
-	return getGVKRFromCR(cr)
-}
-
-func getGVKRFromCR(cr olm.CRDDescription) (group, version, kind, resource string, err error) {
-	version = cr.Version
-	kind = cr.Kind
-
-	gr := strings.SplitN(cr.Name, ".", 2)
-	if len(gr) != 2 {
-		err = fmt.Errorf("couldn't split Custom Resource's name into two: %s", cr.Name)
-		return
-	}
-	resource = gr[0]
-	group = gr[1]
-
-	return
-}
+func GetGVRFromOperator(csv olm.ClusterServiceVersion, cr string) (string, string, string, error) {
+	var group, version, resource string
 
-func GetGVRFromOperator(csv olm.ClusterServiceVersion, cr string) (group, version, resource string, err error) {
 	for _, customresource := range csv.Spec.CustomResourceDefinitions.Owned {
 		custRes := customresource
 		if custRes.Kind == cr {
-			return GetGVRFromCR(&custRes)
+			group, version, resource = GetGVRFromCR(&custRes)
+			return group, version, resource, nil
 		}
 	}
 	return "", "", "", fmt.Errorf("couldn't parse group, version, resource from Operator %q", csv.Name)
 }
 
 // GetGVRFromCR parses and returns the values for group, version and resource
 // for a given Custom Resource (CR).
-func GetGVRFromCR(cr *olm.CRDDescription) (group, version, resource string, err error) {
+func GetGVRFromCR(cr *olm.CRDDescription) (string, string, string) {
+	var group, version, resource string
 	version = cr.Version
 
 	gr := strings.SplitN(cr.Name, ".", 2)
-	if len(gr) != 2 {
```

Comment:
Please add a check before using the indexes to avoid panic

Reply:
I removed it because a CR would always be of the form `<api>/<version>`. It's a check that didn't serve us anything.

Reply:
If someone sets the CR name to a invalid one before the function call, this would panic and stop the execution of the program immediately. It will lead to a bad user experience.

Reply:
Can you think of a scenario or try a scenario in the odo workflow that would cause panic? I don't mind adding this check again if that's the case. It's basically just undoing what I removed from the code.

Reply:
> Can you think of a scenario or try a scenario in the odo workflow that would cause panic?

Any human error and failure to detect the problem on the CI can cause it to panic.

Reply:
> Any human error and failure to detect the problem on the CI can cause it to panic.

Can you please point an odo command that could panic things here?

Reply:
> Can you please point an odo command that could panic things here?

All the commands and code points which call this function are potential candidates.

Reply:
Before this code gets executed as a result of any odo command, we go through https://github.com/openshift/odo/blob/6b03998f5b6b4edb0835d796a90b229e790f7317/pkg/service/service.go#L574

which ensures that if something's not following the `<operator-name>/<cr-name>` convention, odo stops further execution. That's why I don't think odo would hit a panic at this point.

Reply:
> Before this code gets executed as a result of any odo command, we go through

If someone in future tries to call the function outside of this flow and doesn't check against `SplitServiceKindName()`, it might panic. It's always better to prevent such issues before hand. Anyways let's keep it for now.